File: invesdwin-context-python-parent/invesdwin-context-python-runtime-contract/src/test/java/de/invesdwin/context/python/runtime/contract/InputsAndResultsTestDecimal.py
print("getDecimal")
if 'getDecimal' in locals():
	raise Exception("getDecimal already defined!")
getDecimal = putDecimal
getDecimalType = type(getDecimal)
print(getDecimalType)
print(getDecimal)
if(getDecimalType is not float):
	raise Exception("getDecimal not float!")

print("getDecimalVector")
if 'getDecimalVector' in locals():
	raise Exception("getDecimalVector already defined!")
getDecimalVector = putDecimalVector
getDecimalVectorType = type(getDecimalVector[0])
print(getDecimalVectorType)
print(getDecimalVector)
if(getDecimalVectorType is not float):
	raise Exception("getDecimalVector not float!")

print("getDecimalVectorAsList")
if 'getDecimalVectorAsList' in locals():
	raise Exception("getDecimalVectorAsList already defined!")
getDecimalVectorAsList = putDecimalVectorAsList
getDecimalVectorAsListType = type(getDecimalVectorAsList[0])
print(getDecimalVectorAsListType)
print(getDecimalVectorAsList)
if(getDecimalVectorAsListType is not float):
	raise Exception("getDecimalVectorAsList not float!")

print("getDecimalMatrix")
if 'getDecimalMatrix' in locals():
	raise Exception("getDecimalMatrix already defined!")
getDecimalMatrix = putDecimalMatrix
getDecimalMatrixType = type(getDecimalMatrix[0][0])
print(getDecimalMatrixType)
print(getDecimalMatrix)
if(getDecimalMatrixType is not float):
	raise Exception("getDecimalMatrix not float!")

print("getDecimalMatrixAsList")
if 'getDecimalMatrixAsList' in locals():
	raise Exception("getDecimalMatrixAsList already defined!")
getDecimalMatrixAsList = putDecimalMatrixAsList
getDecimalMatrixAsListType = type(getDecimalMatrixAsList[0][0])
print(getDecimalMatrixAsListType)
print(getDecimalMatrixAsList)
if(getDecimalMatrixAsListType is not float):
	raise Exception("getDecimalMatrixAsList not float!")

File: invesdwin-context-python-parent/invesdwin-context-python-runtime-contract/src/test/java/de/invesdwin/context/python/runtime/contract/InputsAndResultsTestDecimal.java
package de.invesdwin.context.python.runtime.contract;

import java.util.ArrayList;
import java.util.List;

import javax.annotation.concurrent.NotThreadSafe;

import org.springframework.core.io.ClassPathResource;

import de.invesdwin.context.integration.script.IScriptTaskEngine;
import de.invesdwin.context.integration.script.IScriptTaskInputs;
import de.invesdwin.context.integration.script.IScriptTaskResults;
import de.invesdwin.util.assertions.Assertions;
import de.invesdwin.util.collections.Arrays;
import de.invesdwin.util.math.decimal.Decimal;

@NotThreadSafe
public class InputsAndResultsTestDecimal {

    private final IScriptTaskRunnerPython runner;

    public InputsAndResultsTestDecimal(final IScriptTaskRunnerPython runner) {
        this.runner = runner;
    }

    public void testDecimal() {
        //putDecimal
        final Decimal putDecimal = new Decimal("123.123");

        //putDecimalVector
        final Decimal[] putDecimalVector = new Decimal[3];
        for (int i = 0; i < putDecimalVector.length; i++) {
            putDecimalVector[i] = Decimal.valueOf((i + 1) + "." + (i + 1));
        }

        //putDecimalVectorAsList
        final List<Decimal> putDecimalVectorAsList = Arrays.asList(putDecimalVector);

        //putDecimalMatrix
        final Decimal[][] putDecimalMatrix = new Decimal[4][];
        for (int row = 0; row < putDecimalMatrix.length; row++) {
            final Decimal[] vector = new Decimal[3];
            for (int col = 0; col < vector.length; col++) {
                vector[col] = Decimal.valueOf((row + 1) + "." + (col + 1));
            }
            putDecimalMatrix[row] = vector;
        }

        //putDecimalMatrixAsList
        final List<List<Decimal>> putDecimalMatrixAsList = new ArrayList<List<Decimal>>(putDecimalMatrix.length);
        for (final Decimal[] vector : putDecimalMatrix) {
            putDecimalMatrixAsList.add(Arrays.asList(vector));
        }

        new AScriptTaskPython<Void>() {

            @Override
            public void populateInputs(final IScriptTaskInputs inputs) {
                inputs.putDecimal("putDecimal", putDecimal);

                inputs.putDecimalVector("putDecimalVector", putDecimalVector);

                inputs.putDecimalVectorAsList("putDecimalVectorAsList", putDecimalVectorAsList);

                inputs.putDecimalMatrix("putDecimalMatrix", putDecimalMatrix);

                inputs.putDecimalMatrixAsList("putDecimalMatrixAsList", putDecimalMatrixAsList);
            }

            @Override
            public void executeScript(final IScriptTaskEngine engine) {
                engine.eval(new ClassPathResource(InputsAndResultsTestDecimal.class.getSimpleName() + ".py",
                        InputsAndResultsTestDecimal.class));
            }

            @Override
            public Void extractResults(final IScriptTaskResults results) {
                //getDecimal
                final Decimal getDecimal = results.getDecimal("getDecimal");
                Assertions.assertThat(putDecimal).isEqualTo(getDecimal);

                //getDecimalVector
                final Decimal[] getDecimalVector = results.getDecimalVector("getDecimalVector");
                Assertions.assertThat(putDecimalVector).isEqualTo(getDecimalVector);

                //getDecimalVectorAsList
                final List<Decimal> getDecimalVectorAsList = results.getDecimalVectorAsList("getDecimalVectorAsList");
                Assertions.assertThat(putDecimalVectorAsList).isEqualTo(getDecimalVectorAsList);

                //getDecimalMatrix
                final Decimal[][] getDecimalMatrix = results.getDecimalMatrix("getDecimalMatrix");
                Assertions.assertThat(putDecimalMatrix).isEqualTo(getDecimalMatrix);

                //getDecimalMatrixAsList
                final List<List<Decimal>> getDecimalMatrixAsList = results
                        .getDecimalMatrixAsList("getDecimalMatrixAsList");
                Assertions.assertThat(putDecimalMatrixAsList).isEqualTo(getDecimalMatrixAsList);
                return null;
            }
        }.run(runner);
    }

}
